tests/contrib/scrapy/test_pipeline_cerberus.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import absolute_import
+import sys
+import os
+import pytest
+from scrapy import Item
+from scrapy.utils.test import get_crawler
+from spidermon.contrib.scrapy.pipelines import ItemValidationPipeline
+from tests.fixtures.items import TestItem, TreeItem
+from tests.fixtures.validators import (
+    cerberus_tree_schema,
+    cerberus_test_schema,
+    test_cerberus_schema_string,
+)
+
+STATS_ITEM_ERRORS = "spidermon/validation/items/errors"
+STATS_TYPES = "spidermon/validation/validators/{}/{}"
+SETTING_CERBERUS = "SPIDERMON_VALIDATION_CERBERUS"
+
+
+def assert_in_cerberus_stats(obj):
+    return "{}".format(STATS_TYPES.format(obj.__name__.lower(), "cerberus"))
+
+
+def the_stats(pipe):
+    return pipe.stats.stats.get_stats()
+
+
+@pytest.mark.parametrize(
+    "item,settings,cases",
+    [
+        pytest.param(
+            TestItem({"url": "example.com"}),
+            {SETTING_CERBERUS: [cerberus_test_schema]},
+            [assert_in_cerberus_stats(Item)],
+            id="processing usual items without errors",
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: {TestItem: cerberus_test_schema}},
+            [assert_in_cerberus_stats(TestItem), STATS_ITEM_ERRORS],
+            id="validator is {} type, loads from a python dict".format(
+                TestItem.__name__
+            ),
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: [f"{os.getcwd()}/contrib/scrapy/schema.json"]},
```

Comment:
You need to run your tests in all supported Python versions. f-strings were introduced in Python 3.6, so this will fail in older version.

Reply:
Resolved now, relative path issue. Not sure if it would run locally without errors.